stablehlo/tests/CheckOps.cpp

```diff
@@ -75,6 +81,53 @@ llvm::Error evalExpectEqOp(const Tensor &lhs, const Tensor &rhs) {
   return llvm::Error::success();
 }
 
+// Fetch a previously serialized filepath given a `probeId` and a `probeDir` for
+// a specified `iteration` value from an `index.csv` metadata file. If no data
+// is found, returns an error.
+static llvm::ErrorOr<std::string> getSerializedTensorPath(StringRef probeId,
+                                                          StringRef probeDir,
+                                                          uint32_t iteration) {
+  if (probeDir.empty()) return llvm::errc::invalid_argument;
+
+  llvm::SmallString<128> instrumentationMetadataFile(probeDir);
+  llvm::sys::path::append(instrumentationMetadataFile,
+                          numpy::kInstrumentationMetadataFilename);
+  std::ifstream metadataFile(instrumentationMetadataFile.str().str());
+
+  if (!metadataFile.is_open()) return llvm::errc::io_error;
+
+  std::string line;
+
+  for (uint32_t match = 0; metadataFile >> line && match <= iteration;
```

Comment:
Fail early when match not found?
```
  for (uint32_t match = 0; metadataFile >> line && match <= iteration;
       ++match) {
    auto pos = line.find(probeId);
    if (pos == std::string::npos) return llvm::errc::bad_address;
    if (match == iteration) return line.substr(pos + probeId.size() + 1);
```

Reply:
Unresolving since the updated commit doesn't seem to address this.

Reply:
Sorry meant to reply to this one. I don't think those proposed changes are equivalent to what the loop is intending to achieve. That is, if we don't find the requested `probeId` in the current line, we break from the loop and return `llvm:errc::bad_address` instead of continuing to check other lines.

Reply:
Got it, thanks!